cmake/developer_package/coverage/coverage.cmake

```diff
@@ -4,210 +4,231 @@
+
+        add_custom_command(OUTPUT ${output_file}
+            COMMAND ${commands}
+            COMMENT "Exclude patterns from report ${OV_COVERAGE_OUTPUT}"
+            DEPENDS ${output_file}${out_suf}
+            VERBATIM)
+    endif()
+
+    add_custom_target(ov_coverage_${OV_COVERAGE_INFO_FILE}_info
                       DEPENDS ${output_file})
-    set_target_properties(ie_coverage_${IE_COVERAGE_INFO_FILE}_info
+    set_target_properties(ov_coverage_${OV_COVERAGE_INFO_FILE}_info
                           PROPERTIES FOLDER coverage)
 endfunction()
 
 #
-# ie_coverage_extract(INPUT <info_file> OUTPUT <output_file> PATTERNS <patterns ...>)
+# ov_coverage_extract(INPUT <info_file> OUTPUT <output_file> PATTERNS <patterns ...>)
 #
-function(ie_coverage_extract)
-    cmake_parse_arguments(IE_COVERAGE "" "INPUT;OUTPUT" "PATTERNS" ${ARGN})
+function(ov_coverage_extract)
+    cmake_parse_arguments(OV_COVERAGE "" "INPUT;OUTPUT" "PATTERNS" ${ARGN})
 
-    set(input_file "${IE_COVERAGE_REPORTS}/${IE_COVERAGE_INPUT}.info")
-    set(output_file "${IE_COVERAGE_REPORTS}/${IE_COVERAGE_OUTPUT}.info")
+    set(input_file "${OV_COVERAGE_REPORTS}/${OV_COVERAGE_INPUT}.info")
+    set(output_file "${OV_COVERAGE_REPORTS}/${OV_COVERAGE_OUTPUT}.info")
 
     set(commands lcov --quiet)
-    foreach(pattern IN LISTS IE_COVERAGE_PATTERNS)
+    foreach(pattern IN LISTS OV_COVERAGE_PATTERNS)
         list(APPEND commands --extract ${input_file} ${pattern})
     endforeach()
     list(APPEND commands --output-file ${output_file})
 
     add_custom_command(OUTPUT ${output_file}
                        COMMAND ${commands}
-                       COMMENT "Generate coverage data ${IE_COVERAGE_OUTPUT}"
+                       COMMENT "Generate coverage data ${OV_COVERAGE_OUTPUT}"
                        DEPENDS ${input_file}
                        VERBATIM)
-    add_custom_target(ie_coverage_${IE_COVERAGE_OUTPUT}_info
+    add_custom_target(ov_coverage_${OV_COVERAGE_OUTPUT}_info
                       DEPENDS ${output_file})
-    set_target_properties(ie_coverage_${IE_COVERAGE_OUTPUT}_info
+    set_target_properties(ov_coverage_${OV_COVERAGE_OUTPUT}_info
                           PROPERTIES FOLDER coverage)
 
-    add_dependencies(ie_coverage_${IE_COVERAGE_OUTPUT}_info ie_coverage_${IE_COVERAGE_INPUT}_info)
+    add_dependencies(ov_coverage_${OV_COVERAGE_OUTPUT}_info ov_coverage_${OV_COVERAGE_INPUT}_info)
+endfunction()
+
+#
+# ov_coverage_genhtml(INFO_FILE <info_file> PREFIX <prefix>)
+#
+function(ov_coverage_genhtml)
+    cmake_parse_arguments(OV_COVERAGE "" "INFO_FILE;PREFIX" "" ${ARGN})
+
+    set(input_file "${OV_COVERAGE_REPORTS}/${OV_COVERAGE_INFO_FILE}.info")
+    set(output_directory "${OV_COVERAGE_REPORTS}/${OV_COVERAGE_INFO_FILE}")
+
+    add_custom_command(OUTPUT "${output_directory}/index.html"
+                       COMMAND [ -s ${input_file}  ] && genhtml ${input_file} --title "${OV_COVERAGE_INFO_FILE}" --legend
```

Comment:
`[ -s ${input_file}  ] && ... || echo "Skip ${input_file}"` is a fix for empty info files